hash.go

```diff
@@ -382,3 +384,67 @@ func (db *RoseDB) HIncrBy(key, field []byte, incr int64) (int64, error) {
 	}
 	return valInt64, nil
 }
+
+// HRandfield return a random field from the hash value stored at key, when called with just
+// the key argument. If the provided count argument is positive, return an array of distinct
+// fields. If called with a negative count, the behavior changes and the command is allowed
+// to return the same field multiple times.
+func (db *RoseDB) HRandfield(key []byte, count interface{}) ([][]byte, error) {
+	db.hashIndex.mu.RLock()
+	defer db.hashIndex.mu.RUnlock()
+
+	countInt, ok := count.(int)
+	if count != nil {
+		if !ok {
+			return [][]byte{}, ErrWrongValueType
+		}
+		if countInt == 0 {
+			return [][]byte{}, nil
+		}
+	}
+	tree, ok := db.hashIndex.trees[string(key)]
+	if !ok {
+		return [][]byte{}, ErrKeyNotFound
+	}
+	var i int
+	keys := make([][]byte, tree.Size())
+	iter := tree.Iterator()
+	for iter.HasNext() {
```

Comment:
you can call HKeys to replace the `for{...}`

Reply:
I certainly have no problem replacing it as the scope of RLock as well.
I will implement it that way.